persist/prepare_options_matcher.go

```diff
@@ -28,12 +28,14 @@ import (
 )
 
 // PrepareOptionsMatcher satisfies the gomock.Matcher interface for PrepareOptions
+// nolint: maligned
 type PrepareOptionsMatcher struct {
-	NsMetadata   namespace.Metadata
-	Shard        uint32
-	BlockStart   time.Time
-	SnapshotTime time.Time
-	FileSetType  FileSetType
+	NsMetadata     namespace.Metadata
```

Comment:
https://github.com/google/go-cmp looks ideal for doing such comparisons at runtime. Mind adopting/ opening a followup issue to track.